core/src/main/java/org/infinispan/commands/read/SizeCommand.java

```diff
@@ -16,15 +19,16 @@
  * @since 4.0
  */
 public class SizeCommand extends AbstractLocalCommand implements VisitableCommand {
-   private final Cache<Object, ?> cache;
+   private final AdvancedCache<?, ?> cache;
 
    public SizeCommand(Cache<Object, ?> cache, long flags) {
       setFlagsBitSet(flags);
+      AdvancedCache<Object, ?> advancedCache = cache.getAdvancedCache();
       if (flags != EnumUtil.EMPTY_BIT_SET) {
-         this.cache = cache.getAdvancedCache().withFlags(EnumUtil.enumArrayOf(flags, Flag.class));
-      } else {
-         this.cache = cache;
+         advancedCache = advancedCache.withFlags(EnumUtil.enumArrayOf(flags, Flag.class));
       }
+      this.cache = advancedCache.getCacheConfiguration().memory().storageType() == StorageType.OBJECT ?
+            advancedCache : advancedCache.withEncoding(IdentityEncoder.class);
```

Comment:
FYI, this logic is part of https://github.com/infinispan/infinispan/pull/5329, to avoid creating new instances if the requested encoding is already in place